internal/reminder/reminder.go

```diff
@@ -53,36 +49,30 @@ type reminder struct {
 	stop     chan struct{}
 	stopOnce sync.Once
 
-	projectListCursor string
-	repoListCursor    map[projectProviderPair]string
+	repositoryCursor uuid.UUID
 
 	ticker *time.Ticker
 
 	eventPublisher message.Publisher
 	eventDBCloser  driverCloser
 }
 
-type projectProviderPair struct {
-	// Exported for gob
-
-	ProjectId uuid.UUID
-	Provider  string
-}
-
 // NewReminder creates a new reminder instance
 func NewReminder(ctx context.Context, store db.Store, config *reminderconfig.Config) (Interface, error) {
-	logger := zerolog.Ctx(ctx)
 	r := &reminder{
-		store:          store,
-		cfg:            config,
-		stop:           make(chan struct{}),
-		repoListCursor: make(map[projectProviderPair]string),
+		store: store,
+		cfg:   config,
+		stop:  make(chan struct{}),
 	}
-	err := r.restoreCursorState(ctx)
+
+	randomRepo, err := r.store.GetRandomRepository(ctx)
```

Comment:
It may generate an out-of-range uuid, which will result in sequential iteration. There is nothing wrong with sequential iteration, but I coded it in a way in which we start from a random valid uuid (theoretically speaking, it is possible to get the first uuid as a random uuid from the db which will result in sequential iteration)

This point of generating a random valid uuid may not be that important, so I'm willing to gen random uuid on application side if that's better (in terms of simplicity)

Reply:
I'm not sure what an out-of-range uuid is -- I'd meant `uuid.NewRandom()`, which generates a valid (v4) UUID.

So, there is one edge case here: we may generate a UUID which is _larger_ than any of the UUIDs in the database (at which point, we should start at the sequentially-first UUID).  I believe that we're using v4 (random) UUIDs, so we should end up with a rougly-even distribution of UUIDs choosing at random.  (If we were using time-sorted UUIDs, picking a random UUID would likely either start at the beginning or the end of the sequence.)